packages/tempus-core-services/src/datastructures/FixedPointDecimal.ts

```
@@ -0,0 +1,139 @@
+import { BigNumber, utils } from 'ethers';
+
+const increasePrecision = (bignum: BigNumber, precision: number): BigNumber => {
+  const multiplicand = BigNumber.from(BigNumber.from(10).pow(precision));
+  return bignum.mul(multiplicand);
+};
+
+const decreasePrecision = (bignum: BigNumber, precision: number): BigNumber => {
+  const divisor = BigNumber.from(BigNumber.from(10).pow(precision));
+  return bignum.div(divisor);
+};
+
+export default class FixedPointDecimal {
+  readonly value: BigNumber;
+  readonly precision: number;
+
+  constructor(value: string | number | BigNumber, precision: number = 18) {
+    if (value instanceof BigNumber) {
+      this.value = value;
+    } else if (value === '.' || value === '') {
```

if we pass `.` or an empty string we probably want to throw an error

---

done with test cases